src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/InteropHelpers.cs

```diff
@@ -591,6 +593,32 @@ internal static unsafe void CleanupVariant(IntPtr pDstNativeVariant)
 #endif
         }
 
+        public static unsafe object InitializeCustomMarshaller(RuntimeTypeHandle pParameterType, RuntimeTypeHandle pMarshallerType, RuntimeTypeHandle pICustomMarshallerType, string cookie, delegate*<string, object> getInstanceMethod)
```

Comment:
`typeof(ICustomMarshaller)` or `EETypePtrOf<ICustomMarshaller>` would work just fine, no need to pass it from the generated IL.

Reply:
Excellent. I was scary to use that because you told me that Type creation is relatively expensive. So I thought this is the case too.